test/liquid/combinators/tags/case_test.exs

```diff
@@ -0,0 +1,173 @@
+defmodule Liquid.Combinators.Tags.CaseTest do
+  use ExUnit.Case
+
+  import Liquid.Helpers
+  alias Liquid.NimbleParser, as: Parser
+
+  test "case using multiples when" do
+    test_combinator(
+      "{% case condition %}{% when 1 %} its 1 {% when 2 %} its 2 {% endcase %}",
+      &Parser.case/1,
+      case: [
+        variable: [parts: [part: "condition"]],
+        whens: [{:when, [1]}, " its 1 ", {:when, [2]}, " its 2 "]
+      ]
+    )
+  end
+
+  test "case using a single when" do
+    test_combinator(
+      "{% case condition %}{% when \"string here\" %} hit {% endcase %}",
+      &Parser.case/1,
+      case: [
+        variable: [parts: [part: "condition"]],
+        whens: [{:when, ["string here"]}, " hit "]
+      ]
+    )
+  end
+
+  test "evaluate variables and expressions" do
+    test_combinator(
+      "{% case a.size %}{% when 1 %}1{% when 2 %}2{% endcase %}",
+      &Parser.case/1,
+      case: [
+        variable: [parts: [part: "a", part: "size"]],
+        whens: [{:when, [1]}, "1", {:when, [2]}, "2"]
+      ]
+    )
+  end
+
+  test "case with a else tag" do
+    test_combinator(
+      "{% case condition %}{% when 5 %} hit {% else %} else {% endcase %}",
+      &Parser.case/1,
+      case: [
+        variable: [parts: [part: "condition"]],
+        whens: [{:when, [5]}, " hit "],
+        else: [" else "]
+      ]
+    )
+  end
+
+  test "when tag with an or condition" do
+    test_combinator(
+      "{% case condition %}{% when 1 or 2 or 3 %} its 1 or 2 or 3 {% when 4 %} its 4 {% endcase %}",
+      &Parser.case/1,
+      case: [
+        variable: [parts: [part: "condition"]],
+        whens: [
+          {:when, [1, {:logical, [:or, 2]}, {:logical, [:or, 3]}]},
+          " its 1 or 2 or 3 ",
+          {:when, [4]},
+          " its 4 "
+        ]
+      ]
+    )
+  end
+
+  test "when with comma's" do
+    test_combinator(
+      "{% case condition %}{% when 1, 2, 3 %} its 1 or 2 or 3 {% when 4 %} its 4 {% endcase %}",
+      &Parser.case/1,
+      case: [
+        variable: [parts: [part: "condition"]],
+        whens: [
```

Comment:
shouldn't `whens` be called `statements` since we're dealing with `case-statements`?